### PR TITLE
[JENKINS-40657] Verify that library objects can be instantiated

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-40657](https://issues.jenkins-ci.org/browse/JENKINS-40657)

Note that this is on top of #80. One thing that did have to change is that the object has to be `Serializable`.

cc @reviewbybees esp @rsandell and also @imod since he opened the issue. =)

### CODE REVIEW

**Comment by imod** [comment]
👍

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java [hunk 1]
```diff
@@ -372,4 +377,73 @@ public boolean apply(FlowNode input) {
...
+                "  def echo(msg) {\n" +
+                "    steps.sh \"echo ${msg}\"\n" +
+                "  }\n" +
+                "}\n");
+        otherRepo.git("add", "src");
+        otherRepo.git("commit", "--message=init");
+        GlobalLibraries.get().setLibraries(Collections.singletonList(
+                new LibraryConfiguration("zot-stuff",
+                        new SCMSourceRetriever(new GitSCMSource(null, otherRepo.toString(), "", "*", "", true)))));
+
```
Comment (rsandell) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java]: _nit_ duplicated code should go into a shared method.

**Comment by rsandell** [comment]
Since this is on top of #80 I'm withholding my bee until my :bug: on that has been addressed ;)

**Comment by abayer** [comment]
#80 is now merged. =)

**Comment by rsandell** [comment]
:bee:

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java [hunk 2]
```diff
@@ -373,6 +378,74 @@ public boolean apply(FlowNode input) {
...
+        otherRepo.init();
+        otherRepo.write("src/org/foo/Zot.groovy", "package org.foo;\n" +
+                "\n" +
+                "class Zot implements Serializable {\n" +
+                "  def steps\n" +
+                "  Zot(steps){\n" +
+                "    this.steps = steps\n" +
+                "  }\n" +
+                "  def echo(msg) {\n" +
+                "    steps.sh \"echo ${msg}\"\n" +
```
Comment (jglick) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java]: More portable to write:

```groovy
step.echo msg
```

**Pull request merged**